### DISCUSSION TITLE
没人想吐槽umi4的codeSplitting吗

### DISCUSSION THREAD

**Discussion by objHua** [umijs/umi#12701] [General]
感觉umi4的codeSplitting策略并不能满足实际生产环境中的需求，很多人都遇到过umi.js bundle过大的问题，想进一步优化，umi又没有提供口子，我想不通，bundle策略做的这么封闭的出发点是什么，难道做企业中后台项目就不用考虑bundle大小吗

**Comment by objHua** [comment]
看了一圈，打算弃了，后续新项目不考虑使用umi4了，宁可多些开发时间搞vite，起码社区活跃度和更新没得比

**Comment by Liuqing650** [comment]
感觉Umi整套设计都在体现一个过度设计了，用起来为啥不能遵守一下正常的开发习惯，啥都喜欢搞特殊化。create-react-app就特别棒，除了他觉得他应该做的，其他的绝不多做。画蛇添足的事情带来的问题和学习成本更复杂

**Reply by Liuqing650** [reply]
这个就是最直观的例子
https://github.com/umijs/umi-next/issues/868#issuecomment-1151088426

**Comment by 5201314999** [comment]
的确有点，自定义口子有点小

**Comment by flgame** [comment]
用webpack的话可以自己写个umi插件，通过插件api.chainWebpack覆盖掉自带的codeSpliting配置。

可能mako不支持，所以没开放
